File: backend/src/main/java/com/jotomo/pomo/task/models/Task.java
package com.jotomo.pomo.task.models;

import com.jotomo.pomo.project.model.Project;
import com.jotomo.pomo.session.model.Session;
import com.jotomo.pomo.task.enums.Priority;
import com.jotomo.pomo.task.enums.Status;
import com.jotomo.pomo.user.model.UserEntity;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
@Entity
@Table(
        name = "tasks",
        uniqueConstraints = {
                @UniqueConstraint(
                        name = "uk_task_owner_title",
                        columnNames = {"owner_id", "project_id", "title"}
                )
        }
)
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, length = 100)
    private String title;

    @Column(length = 1000)
    private String description;

    @Column(nullable = false)
    private boolean archived;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(nullable = false)
    private LocalDateTime updatedAt;

    private LocalDateTime completedAt;

    private LocalDateTime archivedAt;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "owner_id", nullable = false)
    private UserEntity owner;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_task_id")
    private Task parentTask;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Status status;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Priority priority;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "project_id")
    private Project project;

    @Builder.Default
    @ManyToMany(mappedBy = "tasks")
    private List<Session> sessions = new ArrayList<>();
}

File: frontend/src/types/task.ts
export type Priority = "HIGH" | "MEDIUM" | "LOW";

export type Status = "TODO" | "IN_PROGRESS" | "BLOCKED" | "DONE";

export interface Task {
  id: string;
  title: string;
  description: string;
  archived: boolean;
  createdAt: string;
  completedAt?: string | null;
  archivedAt?: string | null;
  ownerId: string;
  parentTaskId?: string | null;
  status: Status;
  priority: Priority;
  projectId?: string | null;
}

export interface CreateTask {
  title: string;
  description: string;
  parentTaskId?: string | null;
  status: Status;
  priority: Priority;
  projectId?: string | null;
}

export interface UpdateTask {
  title?: string;
  description?: string;
  projectId?: string | null;
  parentTaskId?: string | null;
  status?: Status;
  priority?: Priority;
  completedAt?: string | null;
  archived?: boolean;
  archivedAt?: string | null;
}